applications/GeoMechanicsApplication/custom_constitutive/coulomb_yield_surface.cpp

```diff
@@ -61,6 +61,36 @@ Vector CoulombYieldSurface::DerivativeOfFlowFunction(const Vector&, CoulombAvera
     return result;
 }
 
+void CoulombYieldSurface::UpdateSurfaceProperties(double InitialFrictionAngle,
+                                                  double FrictionAngleStrengthFactor,
+                                                  double InitialCohesion,
+                                                  double CohesionStrengthFactor,
+                                                  double InitialDilatancyAngle,
+                                                  double DilatancyAngleStrengthFactor,
+                                                  double kappa)
+{
+    mFrictionAngle = this->UpdateFrictionAngle(InitialFrictionAngle, FrictionAngleStrengthFactor, kappa) *
+                     3.14159265358979323846 / 180;
+    mCohesion = this->UpdateCohesion(InitialCohesion, CohesionStrengthFactor, kappa);
+    mDilatationAngle = this->UpdateDilatancyAngle(InitialDilatancyAngle, DilatancyAngleStrengthFactor, kappa) *
+                       3.14159265358979323846 / 180;
+}
+
+double CoulombYieldSurface::UpdateFrictionAngle(double InitialFrictionAngle, double StrengthFactor, double kappa) const
+{
+    return InitialFrictionAngle + StrengthFactor * kappa;
+}
+
+double CoulombYieldSurface::UpdateCohesion(double InitialCohesion, double StrengthFactor, double kappa) const
+{
+    return InitialCohesion + StrengthFactor * kappa;
```

Comment:
You could also choose to do the conversion to radians here (same holds for the dilatancy angle):
```suggestion
    return MathUtils<>::DegreesToRadians(InitialCohesion + StrengthFactor * kappa);
```